File: log-appenders/nodejs/utilities/logtransporter/src/utils/utility.js
export const escapeString = (inputString) => {
    if (!inputString) {
        return '';
    }

    let escapedString = '';
    for (let i = 0; i < inputString.length; i++) {
        let c = inputString.charAt(i);
        switch (c) {
            case '\\':
                escapedString += '\\\\';
                break;
            case '"':
                escapedString += '\\"';
                break;
            case '\b':
                escapedString += '\\b';
                break;
            case '\f':
                escapedString += '\\f';
                break;
            case '\n':
                escapedString += '\\n';
                break;
            case '\r':
                escapedString += '\\r';
                break;
            case '\t':
                escapedString += '\\t';
                break;
            default:
                if (c < ' ' || c > '~') {
                    let hex = c.charCodeAt(0).toString(16);
                    escapedString += '\\u' + ('0000' + hex).slice(-4);
                } else {
                    escapedString += c;
                }
        }
    }
    return escapedString;
};

File: log-appenders/java/utilities/logtransporter/src/main/java/in/quallit/qlog/logtransporter/utils/Utility.java
package in.quallit.qlog.logtransporter.utils;

/**
 * @author Jigs
 */
public final class Utility {

    private Utility() {}

    public static String escapeString(String inputString) {
        if (inputString == null) {
            return "";
        }

        StringBuilder escapedString = new StringBuilder();
        for (char c : inputString.toCharArray()) {
            switch (c) {
                case '\\':
                    escapedString.append("\\\\");
                    break;
                case '"':
                    escapedString.append("\\\"");
                    break;
                case '\b':
                    escapedString.append("\\b");
                    break;
                case '\f':
                    escapedString.append("\\f");
                    break;
                case '\n':
                    escapedString.append("\\n");
                    break;
                case '\r':
                    escapedString.append("\\r");
                    break;
                case '\t':
                    escapedString.append("\\t");
                    break;
                default:
                    if (c < 32 || c > 126) {
                        escapedString.append(String.format("\\u%04x", (int) c));
                    } else {
                        escapedString.append(c);
                    }
            }
        }
        return escapedString.toString();
    }
}
